apis/tempo/v1alpha1/tempomonolithic_types.go

```diff
@@ -0,0 +1,266 @@
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TempoMonolithicSpec defines the desired state of TempoMonolithic.
+type TempoMonolithicSpec struct {
+	// Storage defines the backend storage configuration
+	//
+	// +kubebuilder:validation:Optional
+	Storage *MonolithicStorageSpec `json:"storage,omitempty"`
+
+	// Ingestion defines the trace ingestion configuration
+	//
+	// +kubebuilder:validation:Optional
+	Ingestion *MonolithicIngestionSpec `json:"ingestion,omitempty"`
+
+	// JaegerUI defines the Jaeger UI configuration
+	//
+	// +kubebuilder:validation:Optional
+	JaegerUI *MonolithicJaegerUISpec `json:"jaegerui,omitempty"`
+
+	// ManagementState defines whether this instance is managed by the operator or self-managed
+	//
+	// +kubebuilder:validation:Optional
+	Management ManagementStateType `json:"management,omitempty"`
+
+	// Observability defines observability configuration for the Tempo deployment
+	//
+	// +kubebuilder:validation:Optional
+	Observability *MonolithicObservabilitySpec `json:"observability,omitempty"`
+
+	// ExtraConfig defines any extra (overlay) configuration for components
+	//
+	// +kubebuilder:validation:Optional
+	ExtraConfig *MonolithicExtraConfigSpec `json:"extraConfig,omitempty"`
+}
+
+// MonolithicStorageSpec defines the storage for the Tempo deployment.
+type MonolithicStorageSpec struct {
+	// Traces defines the backend storage configuration for traces
+	//
+	// +kubebuilder:validation:Required
+	Traces MonolithicTracesStorageSpec `json:"traces"`
+}
+
+// MonolithicTracesStorageSpec defines the traces storage for the Tempo deployment.
+type MonolithicTracesStorageSpec struct {
+	// Backend defines the backend for storing traces. Default: memory
```

Comment:
I am a bit concerned about using the in-memory as default. 

The upstream uses PV as default and the in-memory can be easily overlooked and use significant resources in the cluster.

Reply:
If you're only concerned about OOM situations, the memory counts towards the container resource limit:
> While tmpfs is very fast be aware that, unlike disks, files you write count against the memory limit of the container that wrote them

https://kubernetes.io/docs/concepts/storage/volumes/#emptydir

I'd prefer to keep memory as default, as it's great for quick testing/demos/showcases (and is the default for jaeger all-in-one), and changing it to `pv` is easy (and can/should be mentioned in the docs).